src/prefect/runner/server.py

```diff
@@ -101,13 +117,77 @@ async def get_deployment_router(
             deployment = await client.read_deployment(deployment_id)
             router.add_api_route(
                 f"/deployment/{deployment.id}/run",
-                await _build_endpoint_for_deployment(deployment),
+                await _build_endpoint_for_deployment(deployment, runner),
                 methods=["POST"],
+                name=f"Create flow run for deployment {deployment.name}",
+                description=(
+                    "Trigger a flow run for a deployment as a background task on the"
+                    " runner."
+                ),
+                summary=f"Run {deployment.name}",
             )
 
             # Used for updating the route schemas later on
-            schemas[deployment.name] = deployment.parameter_openapi_schema
-            schemas[deployment.id] = deployment.name
+            schemas[f"{deployment.name}-{deployment_id}"] = (
+                deployment.parameter_openapi_schema
+            )
+            schemas[deployment_id] = deployment.name
+    return router, schemas
+
+
+async def _build_endpoint_for_flow(
+    flow: "Flow", runner: "Runner", entrypoint: str
+) -> Callable:
+    async def _create_flow_run_for_flow(
+        body: Optional[Dict[Any, Any]] = None
+    ) -> JSONResponse:
+        body = body or {}
+        if flow.parameters:
+            try:
+                validate_values_conform_to_schema(body, flow.parameters.dict())
+            except ValueError as exc:
+                raise HTTPException(
+                    status.HTTP_400_BAD_REQUEST,
+                    detail=f"Error creating flow run: {exc}",
+                )
+
+        async with get_client() as client:
+            flow_run = await client.create_flow_run(flow=flow, parameters=body)
+
+        # Schedule the background task
+        runner.execute_in_background(runner.execute_flow_run, flow_run.id, entrypoint)
```

Comment:
Nice! This is essentially having the runner directly run and manage the flow-run invocation, right? This actually looks pretty straightforward. I'd be willing to refactor our run-deployment implementation to cut out the flow-run-create API call (at some point)